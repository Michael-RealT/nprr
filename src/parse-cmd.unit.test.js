"use strict";

const { parseCmd } = require("./parse-cmd.js");

describe.each([
  [[], { scriptName: "", options: [] }],
  [["cmd"], { scriptName: "cmd", options: [] }],
  [["cmd", "--"], { scriptName: "cmd", options: [] }],
  [["cmd", "--", "op1"], { scriptName: "cmd", options: ["op1"] }],
  [["cmd", "--", "op1", "op2"], { scriptName: "cmd", options: ["op1", "op2"] }],
  [["cmd", "op1"], { scriptName: "cmd", options: ["op1"] }],
  [["cmd", "op1", "op2"], { scriptName: "cmd", options: ["op1", "op2"] }],
  [["--"], { scriptName: "", options: [] }],
  [["--", "op1"], { scriptName: "", options: ["op1"] }],
  [["--", "op1", "op2"], { scriptName: "", options: ["op1", "op2"] }]
])("With %j should return %j", (params, expected) => {
  test("Passed as array", () => {
    expect(parseCmd(params)).toEqual(expected);
  });
  test("Passed as string", () => {
    expect(parseCmd(params.join(" "))).toEqual(expected);
  });

  describe("Prefixed by `nprr`", () => {
    let paramsWithNprr = ["nprr", ...params];
    test("Passed as array", () => {
      expect(parseCmd(paramsWithNprr)).toEqual(expected);
    });
    test("Passed as string", () => {
      expect(parseCmd(paramsWithNprr.join(" "))).toEqual(expected);
    });
  });

  describe("Prefixed by `npm run`", () => {
    let paramsWithNpmRun = ["npm", "run", ...params];
    test("Passed as array", () => {
      expect(parseCmd(paramsWithNpmRun)).toEqual(expected);
    });
    test("Passed as string", () => {
      expect(parseCmd(paramsWithNpmRun.join(" "))).toEqual(expected);
    });
  });
});

describe.each([[["/usr/bin/node", "src/index.js"], { scriptName: "", options: [] }]])(
  "With the full argv (%j) should return %j",
  (params, expected) => {
    test("Passed as array", () => {
      expect(parseCmd(params)).toEqual(expected);
    });
    test("Passed as string", () => {
      expect(parseCmd(params.join(" "))).toEqual(expected);
    });
  }
);

it("Should throw an error if called params that are not a string nr an array", () => {
  expect(() => {
    parseCmd({});
  }).toThrowErrorMatchingSnapshot();
});
