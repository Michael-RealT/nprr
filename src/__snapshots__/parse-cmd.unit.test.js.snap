// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should throw an error if called params that are not a string nr an array 1`] = `"[object Object] should be an array or a string"`;
